test/secret/secret_test.go

```diff
@@ -120,6 +120,58 @@ var _ = Describe("secret test", func() {
 
 			// Verify that Pass4SymmKey is updated
 			testenv.VerifySecretsUpdatedOnPod(deployment, testenvInstance, verificationPods, testenv.SecretObject["Pass4SymmKey"], testenv.DecodeBase64(secretObj.Data.Pass4SymmKey))
+
+			// Delete secret key
+			testenv.UpdateSecret(deployment, testenvInstance.GetName(), secretObj, true)
+
+			// Ensure standalone is updating
+			testenv.VerifyStandalonePhase(deployment, testenvInstance, deployment.GetName(), splcommon.PhaseUpdating)
+
+			// Wait for License Master to be in READY status
+			testenv.LicenseMasterReady(deployment, testenvInstance)
+
+			// Wait for Standalone to be in READY status
+			testenv.StandaloneReady(deployment, deployment.GetName(), standalone, testenvInstance)
+
+			// Verify MC Pod is Ready
+			testenv.MCPodReady(testenvInstance.GetName(), deployment)
+
+			// Once system is up after update check each pod for secret key update
+			standaloneSecretName = fmt.Sprintf(testenv.SecretObjectPodName, deployment.GetName(), "standalone", 3)
+			licenseMasterSecretName = fmt.Sprintf(testenv.SecretObjectPodName, deployment.GetName(), "license-master", 3)
+			monitoringConsoleSecretName = fmt.Sprintf(testenv.SecretObjectPodName, testenvInstance.GetName(), "monitoring-console", 3)
+			verificationSecrets = []string{standaloneSecretName, licenseMasterSecretName, monitoringConsoleSecretName}
+
+			// Verify that new HEC TOKEN is created
+			testenv.VerifyNewSecretObjectCreated(deployment, testenvInstance, verificationSecrets, testenv.SecretObject["HecToken"], testenv.DecodeBase64(secretObj.Data.HecToken))
+
+			// Verify that new Admin Password is created
+			testenv.VerifyNewSecretObjectCreated(deployment, testenvInstance, verificationSecrets, testenv.SecretObject["AdminPassword"], testenv.DecodeBase64(secretObj.Data.Password))
+
+			// Verify that new Pass4SymmKey is created
+			testenv.VerifyNewSecretObjectCreated(deployment, testenvInstance, verificationSecrets, testenv.SecretObject["Pass4SymmKey"], testenv.DecodeBase64(secretObj.Data.Pass4SymmKey))
+
+			// Verify that new IdxcSecret is created
+			testenv.VerifyNewSecretObjectCreated(deployment, testenvInstance, verificationSecrets, testenv.SecretObject["IdxcPass4Symmkey"], testenv.DecodeBase64(secretObj.Data.IdxcSecret))
+
+			// Verify that new ShcSecret is created
+			testenv.VerifyNewSecretObjectCreated(deployment, testenvInstance, verificationSecrets, testenv.SecretObject["ShcPass4Symmkey"], testenv.DecodeBase64(secretObj.Data.ShcSecret))
```

Comment:
We should change the method name from `VerifyNewSecretObjectCreated` `VerifySecretOnVersionedSecretObject`

Reply:
modified